lib/topologies/mongos.js

```diff
@@ -286,7 +286,7 @@ Mongos.prototype.connect = function(options) {
   // Create server instances
   var servers = this.s.seedlist.map(function(x) {
     const server = new Server(
-      Object.assign({}, self.s.options, x, {
+      Object.assign({}, self.s.options, x, options, {
```

Comment:
I had assumed we were purposefully not propagating the options down here. Is it possible this will have side effects?

Reply:
spoke about this offline, resolving